pulpanalytics/views.py

```diff
@@ -29,9 +34,28 @@
 ]
 
 
+class LogAndDropData(IntegrityError):
+    def __init__(self, msg):
+        logger.error(msg)
+        super().__init__(msg)
+
+
+def _check_component_version(version):
+    if not version.count(".") == 2:
+        raise LogAndDropData(f"The version string {version} does not have two periods.")
+
+    x, y, z = version.split(".")
+    for item in [x, y, z]:
+        if not item.isdigit():
+            raise LogAndDropData(f"The version string {version} does not only contain numbers.")
+
+
 def _save_components(system, telemetry):
     components = []
     for component in telemetry.components:
+        if not settings.COLLECT_DEV_SYSTEMS:
+            _check_component_version(component.version)
+        components.append(Component(system=system, name=component.name, version=component.version))
```

Comment:
I think this is a duplicate line that you don't want.

Reply:
Conflict resolution is an art.